src/collection/views/CollectionDetail.tsx

```diff
@@ -122,40 +121,26 @@ type CollectionDetailPermissions = Partial<{
 
 type CollectionDetailProps = {
 	id?: string; // Item id when component needs to be used inside another component and the id cannot come from the url (match.params.id)
-	renderDetailLink: (
-		linkText: string | ReactNode,
-		id: string,
-		type: Avo.Core.ContentType,
-		className?: string
-	) => ReactNode;
-	renderSearchLink: (
-		linkText: string | ReactNode,
-		newFilters: FilterState,
-		className?: string
-	) => ReactNode;
-	goToDetailLink: (id: string, type: Avo.Core.ContentType) => void;
-	goToSearchLink: (newFilters: FilterState) => void;
 	enabledMetaData: SearchFilter[];
 };
 
 const CollectionDetail: FunctionComponent<
 	CollectionDetailProps & UserProps & RouteComponentProps<{ id: string }>
-> = ({
-	history,
-	location,
-	match,
-	user,
-	id,
-	renderDetailLink = defaultRenderDetailLink,
-	renderSearchLink = defaultRenderSearchLink,
-	goToDetailLink = defaultGoToDetailLink(history),
-	enabledMetaData = ALL_SEARCH_FILTERS,
-}) => {
+> = ({ history, location, match, user, id, enabledMetaData = ALL_SEARCH_FILTERS }) => {
 	const [t] = useTranslation();
 
 	// State
 	const [collectionId, setCollectionId] = useState(id || match.params.id);
-	const [collection, setCollection] = useState<Avo.Collection.Collection | null>(null);
+
+	const [collectionInfo, setCollectionInfo] = useState<{
+		collection: Avo.Collection.Collection | null;
+		permissions: CollectionDetailPermissions;
+		showLoginPopup: boolean;
+	} | null>(null);
+	const permissions = collectionInfo?.permissions;
+	const showLoginPopup = collectionInfo?.showLoginPopup;
+	const collection = collectionInfo?.collection;
```

Comment:
```suggestion
	const [{collection, permissions, showLoginPopup}, setCollectionInfo] = useState<{
		collection?: Avo.Collection.Collection | null;
		permissions?: CollectionDetailPermissions;
		showLoginPopup?: boolean;
	}>({});
```

Reply:
cool

Reply:
wait, won't that crash when the value is null, since you can't destruct null?